uni/lib/controller/networking/network_router.dart

```diff
@@ -194,7 +308,7 @@ class NetworkRouter {
         }
 
         session
-          ..username = newSession.username
+          ..username = newSession.username // (thePeras): Why is this necessary?
```

Comment:
I think this is necessary because SIGARRA allows login with the username in many formats: upXXXXXXXXX, XXXXXXXXX, etc. As such, after the login is performed, we use the username returned by SIGARRA (which might not be the same we used for login) for everything else from that moment afterwards.

Reply:
Ok, understood